src/uu/mkdir/src/mkdir.rs

```diff
@@ -217,14 +224,33 @@ fn create_dir(path: &Path, recursive: bool, verbose: bool, is_parent: bool) -> U
                 );
             }
             #[cfg(not(windows))]
-            if is_parent {
-                // directories created by -p have permission bits set to '=rwx,u+wx',
-                // which is umask modified by 'u+wx'
-                chmod(path, (!mode::get_umask() & 0o0777) | 0o0300)?;
+            if !path_exists {
+                let acl_perm_bits = get_acl_perm_bits(path);
+                if is_parent {
+                    chmod(path, (!mode::get_umask() & 0o777) | (0o300 | acl_perm_bits))?;
+                } else {
+                    chmod(path, mode | acl_perm_bits)?;
+                }
             }
             Ok(())
         }
+
         Err(_) if path.is_dir() => Ok(()),
         Err(e) => Err(e.into()),
     }
 }
+
+/// Only default acl entries get inherited by objects under the path i.e. if child directories
+/// will have their permissions modified.
+#[cfg(not(windows))]
+fn get_acl_perm_bits(path: &Path) -> u32 {
```

Comment:
we have this function:
https://github.com/uutils/coreutils/blob/8cac375dddb4d4e87519c137c45fa831ac7f2619/src/uucore/src/lib/features/fsxattr.rs#L41
maybe use it ?

Reply:
I tried it but i couldn't figure out how to correctly and reliably interpret the ``Vec<u8>`` value , the key is ``system.posix_acl_default`` for a file when i do ``setfacl -d -m group::rmx test_file`` so that's fine. I just need to reliably parse the byte sequences into extract the permission bits.  

I will try to look through the fsxattr source code in https://github.com/torvalds/linux/blob/master/fs/xattr.c to check how they interpret things. I would definitely appreciate some help or some other resources you can point me to.

Reply:
ok, makes sense
could you please move the function into fsxattr ?

Reply:
> ok, makes sense could you please move the function into fsxattr ?

Okay, will do that.